test/integration_test.py

```diff
@@ -745,18 +750,37 @@ def _test_repository_files(self, catalog: str):
                     response = self._get_url_json(response['Location'])
 
                 response = self._get_url(response['Location'], stream=True)
-                self._validate_fastq_response(response)
+                self._validate_file_response(response, self._file_ext(file))
+
+    def _file_ext(self, file: JSON) -> str:
+        # We believe that the file extension is a more reliable indicator than
+        # the `format` metadata field. Unlike os.path.splitext, this method
+        # preserves multipart extensions, so the extension of "foo.fastq.gz" is
+        # "fastq.gz" instead of "gz"
+        file_name = file['name']
+        _, sep, ext = file_name.lstrip('.').partition('.')
```

Comment:
```python
>>> from pathlib import PurePath
>>> PurePath('.x.y').suffix
'.y'
```

Reply:
Agreed at PL that `suffixes` is the correct method to use.